careers/software-engineer.md

```diff
@@ -4,52 +4,56 @@ The goal of the Software Engineer path is to train engineers to master the funda
+An engineer II is capable of taking on most technical problems and coordinating work spanning several sprints.
 
-Engineer IIs are focused on learning about a specific component or product sub-area and mastering that component.
+**Technical Knowledge:** They have practical experience implementing the core concepts and best practices of software engineering. They are able to write performant, well-abstracted code with minimal help from other engineers. They are able to solve most problems in their domain, but may need help with complex tasks and tasks outside their domain.
 
-**Estimation:** Accountable and reliable to all commitments which they have taken ownership over, both in-office (by completing all work in the Sprint Backlog that they pulled in during Sprint Planning) and when On-Call (by promptly responding to incoming requests)
+**Productivity:** They reliably deliver the work assigned to them in a sprint. When they encounter a complex problem, they leverage the knowledge of their peers to find a suitable solution, even if it means changing the scope of the work. By the PR stage, their code is well-constructed; they do not expect others to catch sloppy mistakes for them.
 
-**Communication:** They communicate well and are capable of delivering feedback to peers and their manager. When given a task with unclear requirements they know how to ask for clarification, and ensure that all assumptions are vetted before work starts to reduce the need for re-work. They understand how their work fits into the larger picture for their team, and use this to identify conflicting requirements to their people leader and product manager. As Engineers learn they will start to participate more in the technical design process, often with guidance from senior engineers. Consistently speaks up in Sprint Planning meetings to ensure complete understanding of each work item. An Engineer II actively seeks out knowledge to supplement the team's weak points and areas of interest, including attending meetups, reading books, etc.
+**Impact:** They can take the lead on delivering work critical to meeting the sprint goal and are capable of independently planning work that multiple engineers can complete in a sprint or two. Instead of just executing on a task as written, they think critically about the problem and ensure that the solution will thoroughly solve the problem.
```

Comment:
For the last sentence, I'd try to make the first part a bit more visible. Instead of stating that they think about the task, how does that actually manifest itself? Do they work with Product/team lead/ DRI to make adjustments? Spark conversations with the team?